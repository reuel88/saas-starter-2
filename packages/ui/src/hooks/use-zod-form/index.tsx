export { useZodForm } from "./use-zod-form";
