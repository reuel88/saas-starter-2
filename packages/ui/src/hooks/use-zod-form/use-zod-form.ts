"use client";

import { useForm, UseFormProps } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
import { TypeOf, ZodSchema } from "zod";

type UseZodFormProps<Z extends ZodSchema> = Exclude<
  UseFormProps<TypeOf<Z>>,
  "resolver"
> & {
  schema: Z;
};

export const useZodForm = <Z extends ZodSchema>({
  schema,
  ...rest
}: UseZodFormProps<Z>) => {
  return useForm({ ...rest, resolver: zodResolver(schema) });
};
