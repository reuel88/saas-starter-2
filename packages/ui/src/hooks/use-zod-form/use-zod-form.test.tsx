import { render } from "@testing-library/react";
import { z } from "zod";
import { useZodForm } from "./use-zod-form";

describe("useZodForm", () => {
  it("render form correctly", () => {
    // Arrange
    const MockComponent = () => {
      const schema = z.object({
        name: z.string(),
        age: z.number(),
      });

      const form = useZodForm({ schema });

      return (
        <form onSubmit={form.handleSubmit(() => {})}>
          <input {...form.register("name")} />
          <input {...form.register("age")} />
        </form>
      );
    };

    // Act
    const { baseElement } = render(<MockComponent />);

    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
