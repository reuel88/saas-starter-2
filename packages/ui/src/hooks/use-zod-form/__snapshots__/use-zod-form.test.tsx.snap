// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useZodForm > render form correctly 1`] = `
<body>
  <div>
    <form>
      <input
        name="name"
      />
      <input
        name="age"
      />
    </form>
  </div>
</body>
`;
